tools/quic/quic_transport_server.py

```diff
@@ -21,13 +21,13 @@
 
 
 class EventHandler:
-    def __init__(self, connection: QuicConnectionProtocol, global_dict: Dict):
+    def __init__(self, connection: QuicConnectionProtocol, global_dict: Dict[str, Any]):
         self.connection = connection
         self.global_dict = global_dict
 
     def handle_client_indication(
             self,
-            origin: str,
+            origin: Optional[str],
```

Comment:
Can you keep this non-optional? Given we excluded the case where `origin` is None in process_client_indication, `origin_string` must not be none there, and hence `origin` must not be None here.

Reply:
There's an error 'Argument 1 to "handle_client_indication" of "EventHandler" has incompatible type "Optional[str]"; expected "str"' at `self.handler.handle_client_indication(origin_string, query)` if this is made non-optional, and at a glance it looks like `origin_string` could be none there, but actually it can't because of the `if origin is None` check. I'll see how I can convince mypy that `origin_string` can't be None there.

Reply:
OK, I've added some asserts which should have the right effect. I'm not sure if there's a prettier way to how that two variables are linked in this way, but asserting works.